modular_skyrat/modules/modular_implants/code/nifsofts/dorms.dm

```diff
@@ -58,3 +58,49 @@
 					/obj/item/clothing/shoes/latex_heels,
 	)
 	purchase_price = 200
+
+/obj/item/disk/nifsoft_uploader/dorms/contract
+	name = "\improper Purpura Contract"
+	loaded_nifsoft = /datum/nifsoft/hypno
+	reusable = TRUE //This is set to true because of how this handles updating laws
+	///What laws will be assigned when using the NIFSoft on someone?
+	var/laws_to_assign = "Law 1: Be nice to others."
+
+/obj/item/disk/nifsoft_uploader/dorms/contract/attempt_software_install(mob/living/carbon/human/target)
+	var/datum/nifsoft/hypno/target_nifsoft = target.find_nifsoft(/datum/nifsoft/hypno)
+	if(target_nifsoft)
+		target_nifsoft.fake_laws = laws_to_assign
+		return TRUE
+
+	. = ..()
+	if(. == FALSE)
+		return FALSE
+
+	target_nifsoft = target.find_nifsoft(/datum/nifsoft/hypno)
+	if(!target_nifsoft)
+		return FALSE
+
+	target_nifsoft.fake_laws = laws_to_assign
+
+/obj/item/disk/nifsoft_uploader/dorms/contract/attack_self(mob/user, list/modifiers)
+	var/new_law = tgui_input_text(user, "Input a new law to add", src, laws_to_assign)
+	if(!new_law)
+		return FALSE
+
+	laws_to_assign = new_law
+	return TRUE
+
+/datum/nifsoft/hypno
+	name = "Purpura Contract"
+	program_desc = "Once installed, the Purpura Contract compells the user to follow the rules stored in the data of the NIFSoft. \n OOC NOTE: This is strictly here for adult roleplay. None of the laws here actually need to be obeyed and you can uninstall this NIFSoft at any time."
+	purchase_price = 0
```

Comment:
A price of 0 is abit cheap compared to other wish items

Reply:
The purchase price variable is the price that you pay when you buy an item through the NIFSoft shop. Since this item can't be bought through the NIFSoft shop, it doesn't have a value assigned to it.